src/internal/target-runner.csx

```diff
@@ -60,6 +57,44 @@ public static class SimpleTargetsTargetRunner
         }
     }
 
+    private static void ValidateDependencies(IDictionary<string, Target> targets)
+    {
+        var missingDependencies = new SortedDictionary<string, ISet<string>>();
+
+        foreach (var targetEntry in targets)
+        {
+            foreach (var dependencyName in targetEntry.Value.Dependencies)
+            {
+                if (!targets.ContainsKey(dependencyName))
+                {
+                    var missingDependenciesForTarget = GetOrAddEmpty(missingDependencies, dependencyName);
+                    missingDependenciesForTarget.Add(targetEntry.Key);
+                }
+            }
+        }
+
+        if (!missingDependencies.Any())
+        {
+            return;
+        }
+
+        var manyDependencies = missingDependencies.Count() > 1;
+        var message = new StringBuilder($"Missing {(manyDependencies ? "dependencies" : "dependency")} detected:");
+        foreach (var missingDependencyEntry in missingDependencies)
+        {
+            var targetsCsv = string.Join(", ", missingDependencyEntry.Value.Select(target => $"\"{target.Replace("\"", "\"\"")}\""));
+            message.Append($@"\r\n  ""{missingDependencyEntry.Key.Replace("\"", "\"\"")}"", required by {targetsCsv}");

```

Comment:
More generally, I wonder if we should really have line breaks in the exception message. These problems are going to get caught very early (on first invocation of a script) so I imagine that the number of missing dependencies is usually going to be very small. Imagine a build script which is run as a part of a larger process, which logs exceptions to a file. One of the things that I've always loathed is when log files do not contain a single log entry per line throughout the file, but rather have some which are spread across a number of lines due to exception messages containing line breaks.

What are your thoughts?

Reply:
One line is fine with me. Some of your earlier comments (months ago) seemed to be leaning in the other direction, not that we explicitly discussed line breaks, but I agree that that appearance of missing targets should be the sort of thing that shows up suddenly, and with very few missing. Stay tuned.